synapse/daemon.py

```diff
@@ -222,12 +222,15 @@ def share(self, name, item):
         except Exception as e:
             logger.exception(f'onTeleShare() error for: {name}')
 
-    def _onDmonFini(self):
+    @s_glob.synchelp
```

Comment:
since this becomes synchelped, do we need to have a separate `afini` inner funciton that we still end up passing into the plex a few lines down?

Reply:
oh, good call.  i was doing that to fix a shutdown race with listenservers .close() stuff, but you're def right.  we can clean that up. :+1: